ice/settings.py

```diff
@@ -27,10 +32,45 @@ class Settings(BaseSettings):
     OUGHT_ICE_AUTO_BROWSER: bool = True
     PAPER_DIR: Path = Path(__file__).parent.parent / "papers"
 
+    # note these attributes are read differently- see [__getattribute__]
+    OPENAI_API_KEY: str = ""
+    OUGHT_INFERENCE_API_KEY: str = ""
+    ELICIT_AUTH_TOKEN: str = ""
+
+    def __get_and_store(self, setting_name: str, prompt: Optional[str] = None) -> str:
+        # We use [__getattribute__] to read these attributes, so that we can
+        # prompt the user for them if they are not already set.
+        if prompt is None:
+            prompt = f"Enter {setting_name}: "
+        if self.__dict__[setting_name] == "":
+            # TODO someday: one way this stinks is that starting the server will
+            # bury this prompt. (the server is started in a background process
+            # and has some log messages). i don't know if there's a clean way
+            # to fix this without re-architecting some stuff.
+            value = input(prompt)
+            setattr(self, setting_name, value)
+            with open(_env_path, "a") as f:
+                # [json.dumps] to escape quotes
+                f.write(f"{setting_name}={json.dumps(value)}\n")
+        return self.__dict__[setting_name]
 
-OUGHT_ICE_DIR = Path(environ.get("OUGHT_ICE_DIR", Path.home() / ".ought-ice"))
+    def __getattribute__(self, __name: str) -> Any:
+        match __name:
+            case "OPENAI_API_KEY":
+                return self.__get_and_store(
+                    "OPENAI_API_KEY",
+                    "Enter OpenAI API key (you can get this from https://beta.openai.com/account/api-keys): ",
+                )
+            case "OUGHT_INFERENCE_API_KEY":
+                return self.__get_and_store("OUGHT_INFERENCE_API_KEY")
+            case "ELICIT_AUTH_TOKEN":
+                return self.__get_and_store("ELICIT_AUTH_TOKEN")
```

Comment:
Looking at where this setting gets used, the check there can be removed and the error message can be moved into instructions here.

Reply:
👍 5f8c3a2d34910a3f69c640225a128c0b984daf7e

Reply:
👍 5783970f18c586c82b8de58fcc361cda69ae96f2